packages/fxa-settings/src/components/FormVerifyTotp/index.tsx

```diff
@@ -2,87 +2,101 @@
-  verifyCode: (code: string) => void;
-};
+// Split inputs are not recommended for accesibility
+// Code inputs should be a single input field
+// See FXA-10390 for details on reverting from split input to single input
 
 const FormVerifyTotp = ({
+  clearBanners,
   codeLength,
+  codeType,
   errorMessage,
-  localizedInputGroupLabel,
+  localizedInputLabel,
   localizedSubmitButtonText,
   setErrorMessage,
   verifyCode,
 }: FormVerifyTotpProps) => {
-  const inputRefs = useRef(
-    Array.from({ length: codeLength }, () =>
-      React.createRef<HTMLInputElement>()
-    )
-  );
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true);
 
-  const [codeArray, setCodeArray] = useState<CodeArray>(new Array(codeLength));
-  const [isSubmitting, setIsSubmitting] = useState(false);
+  const ftlMsgResolver = useFtlMsgResolver();
 
-  const stringifiedCode = codeArray.join('');
+  const { handleSubmit, register } = useForm<VerifyTotpFormData>({
+    mode: 'onBlur',
+    criteriaMode: 'all',
+    defaultValues: {
+      code: '',
+    },
+  });
 
-  const isFormValid = stringifiedCode.length === codeLength && !errorMessage;
-  const isSubmitDisabled = isSubmitting || !isFormValid;
-
-  const handleSubmit = async (e: React.FormEvent) => {
-    e.preventDefault();
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (errorMessage) {
+      setErrorMessage('');
+    }
+    // only accept characters that match the code type (numeric or alphanumeric)
+    // strip out any other characters
+    const filteredCode = e.target.value.replace(
+      codeType === 'numeric' ? /[^0-9]/g : /[^a-zA-Z0-9]/g,
+      ''
+    );
+    e.target.value = filteredCode;
+    e.target.value.length > 0
+      ? setIsSubmitDisabled(false)
+      : setIsSubmitDisabled(true);
+  };
 
-    if (!isSubmitDisabled) {
-      setIsSubmitting(true);
-      await verifyCode(stringifiedCode);
-      setIsSubmitting(false);
+  const onSubmit = async ({ code }: VerifyTotpFormData) => {
+    clearBanners && clearBanners();
+    setIsSubmitDisabled(true);
+    // Only submit the code if it is the correct length
+    // Otherwise, show an error message
+    if (code.length !== codeLength) {
+      setErrorMessage(
+        getLocalizedErrorMessage(ftlMsgResolver, AuthUiErrors.INVALID_OTP_CODE)
+      );
+    } else if (!isSubmitDisabled) {
+      await verifyCode(code);
     }
+    setIsSubmitDisabled(false);
   };
 
   return (
-    <>
-      {errorMessage && <Banner type={BannerType.error}>{errorMessage}</Banner>}
-      <form
-        noValidate
-        className="flex flex-col gap-4 my-6"
-        onSubmit={handleSubmit}
+    <form
+      noValidate
+      className="flex flex-col gap-4 my-6"
+      onSubmit={handleSubmit(onSubmit)}
+    >
+      {/* Using `type="text" inputmode="numeric"` shows the numeric keyboard on mobile
+      and strips out whitespace on desktop, but does not add an incrementer. */}
+      <InputText
```

Comment:
Something that crossed my mind, probably not for this PR, but submitting the form when we detect that the text was pasted in. It would elminate one extra button click.

Reply:
Oooh I like this idea! I'll mention to Eduardo to see where we can keep track :)